GUI/QtQuick/QtQuickAdapter.cpp

```diff
@@ -688,15 +690,21 @@ ProcessingResult QtQuickAdapter::processWallets(const Envelope &env)
       emit walletsLoaded(msg.ready());
       logger_->debug("[{}] loaded {} wallet[s]", __func__, msg.ready());
       {
-         if (settingsController_->hasParam(ApplicationSettings::Setting::SelectedWallet)) {
-            const int lastIdx = settingsController_->getParam(ApplicationSettings::Setting::SelectedWallet).toInt();
-            if ((lastIdx >= 0) && (lastIdx < nWalletsLoaded_)) {
-               walletSelected(lastIdx);
-            }
-            else if (nWalletsLoaded_ > 0) {
-               walletSelected(0);
+         if (!createdWallet_) {
+            if (settingsController_->hasParam(ApplicationSettings::Setting::SelectedWallet)) {
+               const int lastIdx = settingsController_->getParam(ApplicationSettings::Setting::SelectedWallet).toInt();
+               if ((lastIdx >= 0) && (lastIdx < nWalletsLoaded_)) {
+                  walletSelected(lastIdx);
+               }
+               else if (nWalletsLoaded_ > 0) {
+                  walletSelected(0);
+               }
             }
          }
+         else {
+            createdWallet_ = false;
+            emit requestWalletSelection(nWalletsLoaded_ - 1);
```

Comment:
Newly created wallet is not always the last one in the list. In created_wallet message there's a wallet_id, field, you can use it to detect the correct index.